modules/fedimint-gw-client/src/pay.rs

```diff
@@ -0,0 +1,17 @@
+/// State machine that pays a lightning invoice on behalf of a federation user.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    PayInvoice -- validate outgoing contract --> Canceled
```

Comment:
Do we assume we already have the contract? Otherwise we might prepend a step that first fetches it? That would likely be modeled as one step with validation in the code, but conceptually seems like 2.

Reply:
Yes I'm making that assumption here. I can split that into two states

Reply:
Made into two states